File: api-xmlsignature/csharp/test/src/dev.esya.api.xmlsignature/unit/signerHelpers/EPfxSignerManager.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;
using tr.gov.tubitak.esya.api.xmlsignature.unit.test;
using tr.gov.tubitak.uekae.esya.api.asn.x509;
using tr.gov.tubitak.uekae.esya.api.common.crypto;

namespace tr.gov.tubitak.uekae.esya.api.xmlsignature.unit.pfxsigner
{
    public class EPfxSignerManager : ISignerManager
    {

        Dictionary<int,EPfxSigner> pfxSignerMap = new Dictionary<int, EPfxSigner>(); 
        private static EPfxSignerManager instance = null;

        public static string getPfxFileDirPath()
        {
            return "T:\\api-parent\\resources\\ug\\pfx\\test-suite\\";
        }

        private string pfxFileDirPath;
        public static EPfxSignerManager getInstance(string pfxFileDirPath)
        {
            if (instance == null)
            {
                if (pfxFileDirPath == null)
                    pfxFileDirPath = getPfxFileDirPath();
                instance = new EPfxSignerManager(pfxFileDirPath);
            }
            return instance;
        }

        private EPfxSignerManager(string pfxFileDirPath)
        {
            this.pfxFileDirPath = pfxFileDirPath;
        }

        private EPfxSigner constructPfxSigner(int slot)
        {
            EPfxSigner retPfxSigner = null;
            switch (slot)
            {
                case 0:
                    retPfxSigner = new EPfxSigner(UnitTestParameters.SIGNATURE_ALGORITHM, pfxFileDirPath + UnitTestParameters.PFX_FILE, UnitTestParameters.PFX_PASSWORD);
                    break;
                case 1:
                    retPfxSigner = new EPfxSigner(UnitTestParameters.SIGNATURE_ALGORITHM, pfxFileDirPath + UnitTestParameters.PFX_2_FILE, UnitTestParameters.PFX_2_PASSWORD);
                    break;
                case 2:
                    retPfxSigner = new EPfxSigner(UnitTestParameters.SIGNATURE_ALGORITHM, pfxFileDirPath + UnitTestParameters.PFX_3_FILE, UnitTestParameters.PFX_3_PASSWORD);
                    break;
                case 3:
                    retPfxSigner = new EPfxSigner(UnitTestParameters.SIGNATURE_ALGORITHM, pfxFileDirPath + UnitTestParameters.PFX_REVOKED_FILE, UnitTestParameters.PFX_REVOKED_PASSWORD);
                    break;
            }
            return retPfxSigner;
        }

        public BaseSigner getSigner(int slot)
        {
            if (!pfxSignerMap.ContainsKey(slot))
            {
                EPfxSigner ePfxSigner = constructPfxSigner(slot);
                pfxSignerMap.Add(slot,ePfxSigner);
            }
            EPfxSigner pfxSigner = pfxSignerMap[slot];
            return pfxSigner;
        }


        public BaseSigner getSigner(int signSlotNo, int certSlotNo)
        {
            ECertificate signatureCertificate = getSigningCertificate(certSlotNo);
            EPfxSigner signPfxSigner = (EPfxSigner) getSigner(signSlotNo);
            signPfxSigner.setSignRequestCertificate(signatureCertificate);
            return signPfxSigner;
        }

        public ECertificate getSigningCertificate(int slot)
        {
            EPfxSigner ePfxSigner = (EPfxSigner) getSigner(slot);
            return ePfxSigner.getSigningCertificate();
        }

        public void logout(int slot)
        {
            
        }
    }
}

File: api-xmlsignature/testsrc/dev/esya/api/xmlsignature/legacy/signerHelpers/EPfxSignerManager.java
package dev.esya.api.xmlsignature.legacy.signerHelpers;

import dev.esya.api.xmlsignature.legacy.UnitTestParameters;
import tr.gov.tubitak.uekae.esya.api.asn.x509.ECertificate;
import tr.gov.tubitak.uekae.esya.api.common.crypto.BaseSigner;
import tr.gov.tubitak.uekae.esya.api.signature.util.PfxSigner;

import java.io.File;
import java.net.URL;
import java.util.HashMap;
import java.util.Map;

/**
 * Created with IntelliJ IDEA.
 * User: yavuz.kahveci
 * Date: 30.01.2013
 * Time: 13:31
 * To change this template use File | Settings | File Templates.
 */
public class EPfxSignerManager implements ISignerManager
{
    Map<Integer, PfxSigner> pfxSignerMap = new HashMap<Integer, PfxSigner>();
    private static EPfxSignerManager instance = null;
    private static String pfxFileDirPath;

    public static String getPfxFileDirPath()
    {
        URL root = EPfxSignerManager.class.getResource("../../../../../../../../../../");
        String classPath = root.getPath();
        File binDir = new File(classPath);
        String rootDir = binDir.getParentFile().getParent();
        String pfxDirPath = rootDir+"/docs/certstore/";
        return pfxDirPath;
    }

    private EPfxSignerManager(String aPfxFileDirPath)
    {
        pfxFileDirPath = aPfxFileDirPath;
    }

    public static EPfxSignerManager getInstance(String aPfxFileDirPath)
    {
        if(instance == null)
        {
            if(aPfxFileDirPath == null)
            {
                pfxFileDirPath = getPfxFileDirPath();
            }
            instance = new EPfxSignerManager(pfxFileDirPath);
        }
        return instance;
    }

    private PfxSigner constructPfxSigner(int slot)
    {
        PfxSigner retPfxSigner = null;
        switch (slot)
        {
            case 0: retPfxSigner = new PfxSigner(UnitTestParameters.SIGNATURE_ALG, pfxFileDirPath + UnitTestParameters.PFX_FILE, UnitTestParameters.PFX_PASSWORD.toCharArray()); break;
            case 1: retPfxSigner = new PfxSigner(UnitTestParameters.SIGNATURE_ALG, pfxFileDirPath + UnitTestParameters.PFX_2_FILE, UnitTestParameters.PFX_2_PASSWORD.toCharArray()); break;
            case 2: retPfxSigner = new PfxSigner(UnitTestParameters.SIGNATURE_ALG, pfxFileDirPath + UnitTestParameters.PFX_3_FILE, UnitTestParameters.PFX_3_PASSWORD.toCharArray()); break;
            case 3: retPfxSigner = new PfxSigner(UnitTestParameters.SIGNATURE_ALG, pfxFileDirPath + UnitTestParameters.PFX_REVOKED_FILE, UnitTestParameters.PFX_REVOKED_PASSWORD.toCharArray()); break;
            default: retPfxSigner = null; break;
        }
        return retPfxSigner;
    }

    public BaseSigner getSigner(int slot)
    {
        if (!pfxSignerMap.containsKey(slot))
        {
            PfxSigner ePfxSigner = constructPfxSigner(slot);
            pfxSignerMap.put(slot,ePfxSigner);
        }
        PfxSigner pfxSigner = pfxSignerMap.get(slot);
        return pfxSigner;
    }

    public BaseSigner getSigner(int signSlotNo, int certSlotNo)
    {
        //ECertificate signatureCertificate = getSigningCertificate(certSlotNo);
        PfxSigner signPfxSigner = (PfxSigner) getSigner(signSlotNo);
        //signPfxSigner.setSignRequestCertificate(signatureCertificate);
        return signPfxSigner;
    }

    public ECertificate getSigningCertificate(int slot)
    {
        PfxSigner ePfxSigner = (PfxSigner) getSigner(slot);
        return ePfxSigner.getSignersCertificate();
    }

    public void logout(int slot)
    {
        //To change body of implemented methods use File | Settings | File Templates.
    }
}
